examples/dataset/classification/classification/binary/upload_results.py

```diff
@@ -0,0 +1,138 @@
+from classification.binary.constants import POSITIVE_LABEL
+
+import kolena
+from kolena.dataset import fetch_dataset
+from kolena.dataset import test
+from kolena.workflow._datatypes import _get_full_type
+from kolena.workflow.annotation import ScoredClassificationLabel
+
+MODELS = ["resnet50v2", "inceptionv3"]
+eval_config = {"threshold": 0.5}
+id_fields = ["locator"]
+
+
+def metrics(score, ground_truth_label) -> Dict[str, Any]:
+    prediction = ScoredClassificationLabel(label=NEGATIVE_LABEL, score=score)
+    if score >= eval_config["threshold"]:
+        prediction = ScoredClassificationLabel(label=POSITIVE_LABEL, score=score)
+
+    threshold = eval_config["threshold"]
+    if threshold is None:
+        threshold = 0.5
+
+    is_positive_prediction = prediction.score >= threshold
+    is_positive_sample = ground_truth_label == prediction.label
+
+    return {
+        "classification_label": prediction.label,
+        "classification_score": prediction.score,
+        "threshold": threshold,
+        "is_correct": is_positive_prediction == is_positive_sample,
+        "is_TP": is_positive_sample and is_positive_prediction,
+        "is_FP": not is_positive_sample and is_positive_prediction,
+        "is_FN": is_positive_sample and not is_positive_prediction,
+        "is_TN": not is_positive_sample and not is_positive_prediction,
+    }
+
+
+def to_kolena_inference(score, multiclass: bool = False) -> List[Dict[str, Any]]:
+    # treating this as multiclass classification problem (i.e. both positive and negative labels are
+    # equally important) — prediction for both labels are required.
+
+    label = POSITIVE_LABEL
+    score = score
+    kolena_label = ScoredClassificationLabel(label=label, score=score)
+    result = [
+        {
+            "label": label,
+            "score": score,
+            "data_type": _get_full_type(kolena_label),
+        },
+    ]
+    if multiclass:
+        neg_label = NEGATIVE_LABEL
+        neg_score = 1 - score
+        neg_kolena_label = ScoredClassificationLabel(label=neg_label, score=neg_score)
+        result.append(
+            {
+                "label": neg_label,
+                "score": neg_score,
+                "data_type": _get_full_type(neg_kolena_label),
+            },
+        )
+    return result
+
+
+def upload_results(model_name: str, dataset: str, multiclass: bool) -> None:
+    df_results = pd.read_csv(f"s3://{BUCKET}/{DATASET}/results/predictions_{model_name}.csv")
+    dataset_df = fetch_dataset(dataset)
+    df_results = df_results.merge(dataset_df, how="left", on=id_fields)
+
+    raw_inference = df_results["prediction"].apply(lambda x: to_kolena_inference(x, multiclass))
+    raw_inference.name = "raw_inference"
+
+    eval_result = df_results.apply(lambda x: pd.Series(metrics(x.prediction, x.label)), axis=1)
+    eval_result.columns = [
+        "classification_label",
+        "classification_score",
+        "threshold",
+        "is_correct",
+        "is_TP",
```

Comment:
If it's multiclass, which label becomes "positive" here?

Reply:
Dog will always be the positive one, but when it is multiclass, it is more like 1st class (dog) and 2nd class (cat)